File: module1/Case Study/Customer.js
class Customer {
    constructor(name,cmnd,birthday,email,address,typeCustomer,typeRoom,discount,quantityIncluded,rentDay,typeService) {
        this.name = name;
        this.cmnd = cmnd;
        this.birthday = birthday;
        this.email = email;
        this.address = address;
        this.typeCustomer = typeCustomer;
        this.typeRoom = typeRoom;
        this.discount = discount;
        this.quantityIncluded = quantityIncluded;
        this.rentDay = rentDay;
        this.typeService = typeService;
    }
    getName(){
        return this.name;
    }
    getCmnd(){
        return this.cmnd;
    }
    getBirthday(){
        return this.birthday;
    }
    getEmail(){
        return this.email;
    }
    getAddress(){
        return this.address;
    }
    geTypeCustomer(){
        return this.typeCustomer;
    }
    getTypeRoom(){
        return this.typeRoom;
    }
    getDiscount(){
        return this.discount;
    }
    getQuantityIncluded(){
        return this.quantityIncluded;
    }
    getRentDay(){
        return this.rentDay;
    }
    getTypeService(){
        return this.typeService;
    }
    setName(name){
         this.name = name;
    }
    setCmnd(cmnd){
         this.cmnd = cmnd;
    }
    setBirthday(birthday){
         this.birthday = birthday;
    }
    setEmail(email){
         this.email = email;
    }
    setAddress(address){
        this.address = address;
    }
    setTypeCustomer(typecustomer){
        this.typeCustomer = typecustomer;
    }
    setTypeRoom(typeroom){
        this.typeRoom = typeroom;
    }
    setDiscount(discount){
        this.discount = discount;
    }
    setQuantityIncluded(quantityincluded){
        this.quantityIncluded = quantityincluded;
    }
    setRentDay(rentday){
        this.rentDay = rentday;
    }
    setTypeService(typeservice){
        this.typeService = typeservice;
    }
    totalPay(){
        let money = 0;
        if (this.getTypeService() === "Villa"){
            money = 500;
        } else if (this.getTypeService() === "House"){
            money = 300;
        }else if (this.getTypeService() === "Room"){
            money = 200;
        }
        return this.getRentDay()*money*(1-this.getDiscount()/100);
    }
}

File: FuramaResort/src/Models/Customer.java
package Models;

public class Customer {
    private String name;
    private String birthday;
    private String gender;
    private long phone;
    private long identityCard;
    private String email;
    private String typeCustomer;
    private String address;
    private Services services;

    public Customer() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public String isGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public long getPhone() {
        return phone;
    }

    public void setPhone(long phone) {
        this.phone = phone;
    }

    public long getIdentityCard() {
        return identityCard;
    }

    public void setIdentityCard(long identityCard) {
        this.identityCard = identityCard;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTypeCustomer() {
        return typeCustomer;
    }

    public void setTypeCustomer(String typeCustomer) {
        this.typeCustomer = typeCustomer;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Services getServices() {
        return services;
    }

    public void setServices(Services services) {
        this.services = services;
    }

    public void showInfor() {
    }
    public int getYearBirthday() {
        String temp = "";
        for (int i = birthday.length() - 4; i < birthday.length(); i++) {
            temp += birthday.charAt(i);
        }
        return new Integer(temp);
    }
    @Override
    public String toString() {
        return "Customer{" +
                "name='" + name + '\'' +
                ", birthday='" + birthday + '\'' +
                ", gender=" + gender +
                ", phone=" + phone +
                ", identityCard=" + identityCard +
                ", email='" + email + '\'' +
                ", typeCustomer='" + typeCustomer + '\'' +
                ", address='" + address + '\'' +
                ", services=" + services +
                '}';
    }
}
